apps/ledger-live-desktop/src/renderer/families/bitcoin/StepReceiveFunds.tsx

```diff
@@ -0,0 +1,359 @@
+import invariant from "invariant";
+import React, { useEffect, useRef, useCallback, useState } from "react";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import { getMainAccount, getAccountName } from "@ledgerhq/live-common/account/index";
+import TrackPage from "~/renderer/analytics/TrackPage";
+import ErrorDisplay from "~/renderer/components/ErrorDisplay";
+import { DisconnectedDevice } from "@ledgerhq/errors";
+import { Trans } from "react-i18next";
+import styled from "styled-components";
+import useTheme from "~/renderer/hooks/useTheme";
+import { urls } from "~/config/urls";
+import { openURL } from "~/renderer/linking";
+import Box from "~/renderer/components/Box";
+import Button from "~/renderer/components/Button";
+import Text from "~/renderer/components/Text";
+import Ellipsis from "~/renderer/components/Ellipsis";
+import ReadOnlyAddressField from "~/renderer/components/ReadOnlyAddressField";
+import LinkWithExternalIcon from "~/renderer/components/LinkWithExternalIcon";
+import LinkShowQRCode from "~/renderer/components/LinkShowQRCode";
+import SuccessDisplay from "~/renderer/components/SuccessDisplay";
+import Receive2NoDevice from "~/renderer/components/Receive2NoDevice";
+import { renderVerifyUnwrapped } from "~/renderer/components/DeviceAction/rendering";
+import { AccountLike } from "@ledgerhq/types-live";
+import { track } from "~/renderer/analytics/segment";
+import Modal from "~/renderer/components/Modal";
+import Alert from "~/renderer/components/Alert";
+import ModalBody from "~/renderer/components/Modal/ModalBody";
+import QRCode from "~/renderer/components/QRCode";
+import { getEnv } from "@ledgerhq/live-common/env";
+import AccountTagDerivationMode from "~/renderer/components/AccountTagDerivationMode";
+import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
+import { useDispatch } from "react-redux";
+import { openModal } from "~/renderer/actions/modals";
+import { StepProps } from "~/renderer/modals/Receive/Body";
+import { LOCAL_STORAGE_KEY_PREFIX } from "~/renderer/modals/Receive/steps/StepReceiveStakingFlow";
+
+const Separator = styled.div`
+  border-top: 1px solid #99999933;
+  margin: 50px 0;
+`;
+const Separator2 = styled.div`
+  border-top: 1px solid #99999933;
+  margin-top: 50px;
+`;
+const QRCodeWrapper = styled.div`
+  border: 24px solid white;
+  height: 208px;
+  width: 208px;
+  background: white;
+`;
+const AlertBoxContainer = styled.div`
+  margin-top: 20px;
+`;
+const Receive1ShareAddress = ({
```

Comment:
Yeah it's a lot of existing code, because one of our external dev has done it for `hedera` and I simply copy paste code from the original receive flow.

I can try to make it work cleaner tho but will not be a quickwin, I have a lot of thing I'm afraid of touching but will give it a shot

Reply:
Did it, it's indeed better